frontend/src/components/search/SearchBar.jsx

```diff
@@ -0,0 +1,47 @@
+// NPM Packages
+import React, { useEffect, useState } from "react";
+import { useHistory } from "react-router-dom";
+import UserApi from "../../api/UserApi";
+
+export default function SearchBar(){
+    // Local State
+    const [query, setQuery] = useState("");
+    const [users, setUsers] = useState([]);
+
+    // Constants
+    const history = useHistory();
+
+    useEffect(() => {
+      UserApi.getAllUsers()
+      .then(({ data }) => {
+        setUsers(data);
+      })
+      .catch((err) => console.error(err));
+    }, [setUsers]);
+
+    // Methods
+    function searchUser(event) {
+        event.preventDefault();
+        //Find out the user by the Id
+        const userId = users.filter((user) => user.name.toUpperCase().match(query.toUpperCase()));
+        history.push(`/profile/${userId[0].id}`);
```

Comment:
Could we add some error handling if the profile doesn't exist and show an alert? 
<img width="1031" alt="Screenshot 2021-05-17 at 10 58 33" src="https://user-images.githubusercontent.com/46404690/118462471-feefaa00-b6fe-11eb-9fa7-8e19df1368eb.png">

Reply:
a suggestion, before history.push you can have if-else and check that userId is not empty, if it is do alert ("no such user found")

Reply:
The error has been handled with alert